shared/actions/kbfs/platform.desktop.js

```diff
@@ -4,63 +4,74 @@ import {shell} from 'electron'
-  if (!path.startsWith(Constants.defaultKBFSPath)) {
-    console.warn(`ERROR: openInKBFS requires ${Constants.defaultKBFSPath} prefix`)
+function openInWindows (openPath: string = Constants.defaultKBFSPath): AsyncAction {
+  openPath = path.resolve(openPath) // Path resolve removes any ..
+  if (!openPath.startsWith(Constants.defaultKBFSPath)) {
+    console.warn(`openInKBFS requires ${Constants.defaultKBFSPath} prefix: ${openPath}`)
     return () => {}
   }
-  path = path.slice(Constants.defaultKBFSPath.length)
+  openPath = openPath.slice(Constants.defaultKBFSPath.length)
 
   return (dispatch, getState) => new Promise((resolve, reject) => {
     const state = getState()
     const kbfsPath = state.config.kbfsPath
 
     // On windows the path isn't /keybase
     // We can figure it out by looking at the extendedConfig though
-    if (process.platform === 'win32') {
-      if (kbfsPath === Constants.defaultKBFSPath) {
-        const extendedConfig = Promise.resolve(state.config.extendedConfig)
+    if (kbfsPath === Constants.defaultKBFSPath) {
+      const extendedConfig = Promise.resolve(state.config.extendedConfig)
 
-        extendedConfig.then(extendedConfig => {
-          const kbfsClients = extendedConfig.Clients.filter(c => c.clientType === Common.ClientType.kbfs)
-          if (kbfsClients.length !== 1) {
-            return Promise.reject("There isn't exactly one kbfs client")
-          }
+      extendedConfig.then(extendedConfig => {
+        const kbfsClients = extendedConfig.Clients.filter(c => c.clientType === Common.ClientType.kbfs)
+        if (kbfsClients.length !== 1) {
+          return Promise.reject("There isn't exactly one kbfs client")
+        }
 
-          // Hacky Regex to find a mount point on windows matches anything like foobar:\ or K:\
-          const kbfsPath = kbfsClients[0].argv.filter(arg => arg.search(/.*:\\?$/) === 0)[0]
+        // Hacky Regex to find a mount point on windows matches anything like foobar:\ or K:\
+        const kbfsPath = kbfsClients[0].argv.filter(arg => arg.search(/.*:\\?$/) === 0)[0]
 
-          if (!kbfsPath) {
-            return Promise.reject("Couldn't figure out kbfs path from argv")
-          }
+        if (!kbfsPath) {
+          return Promise.reject("Couldn't figure out kbfs path from argv")
+        }
 
-          return Promise.resolve(kbfsPath + '\\')
-        }).then(kbfsPath => {
-          dispatch({type: Constants.changeKBFSPath, payload: {path: kbfsPath}})
-          open(kbfsPath, path)
-        }).catch(e => {
-          console.warn('Error in parsing kbfsPath: ', e)
-        })
-      } else {
-        open(kbfsPath, path)
-      }
+        return Promise.resolve(kbfsPath + '\\')
+      }).then(kbfsPath => {
+        dispatch({type: Constants.changeKBFSPath, payload: {path: kbfsPath}})
+        open(path.resolve(kbfsPath, openPath))
+      }).catch(e => {
+        console.warn('Error in parsing kbfsPath:', e)
```

Comment:
I'm reluctant to change this part of the code cause I was having trouble testing. Should I make an issue to clean this method up? It could use a little love but it's a bit out of scope.